### PR TITLE
[iOS] Fix build failure on Xcode 16 Beta

### PR DESCRIPTION
**Pull request by alantoa**
# Why
Hey, I noticed a build issue with Xcode 16 beta. It's because in iOS 18 and macOS 15, the `CNAuthorizationStatus` enum includes a new constant called [limited](https://developer.apple.com/documentation/contacts/cnauthorizationstatus/limited). However, the current version is missing it, which causes a build error with Xcode 16.

CleanShot 2024-06-20 at 23 58 24@2x

# How

Add the new enum type to fix it. 

# Test Plan

 

I texted it to my project and it works well. 

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-contacts/CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 - Fixed an issue where the `requestPermissionsAsync` promise throws when denying access to contacts on iOS. ([#29529](https://github.com/expo/expo/pull/29529) by [@jp1987](https://github.com/jp1987))
 - Fixed an issue where the `presentFormAsync` promise doesn't resolve when the form is closed on Android. ([#29201](https://github.com/expo/expo/pull/29201) by [@jp1987](https://github.com/jp1987))
 - Fixed an issue where the `presentContactPickerAsync` promise doesn't resolve when using the Android back button. ([#29202](https://github.com/expo/expo/pull/29202) by [@jp1987](https://github.com/jp1987))
-
+- Fixed an issue with Xcode 16 Beta builds. ([#29907](https://github.com/expo/expo/pull/29907) by [@alantoa](https://github.com/alantoa))
```
Comment (tsapeta) [inline packages/expo-contacts/CHANGELOG.md:8]: Oops, looks like the changelog was messed up with the entries above – they should be in "bug fixes" section 😅  Could you move them there (including your entry)?

Comment (alantoa) [reply packages/expo-contacts/CHANGELOG.md]: ah yeah, sorry. just moved it.

**Comment by alantoa** [comment]
@alanjhughes, got it, thanks! I didn't realize that PR had already been opened.

**Comment by tsapeta** [comment]
I think it can be closed now, as #29639 has been merged (and as opposed to this PR, it's fully compatible with older Xcode versions).

**Pull request closed without merging**